lotto-vanilla/App.js

```diff
@@ -0,0 +1,131 @@
+import CoinInput from "./src/components/CoinInput";
+import Header from "./src/components/Header";
+import LottoInput from "./src/components/LottoInput";
+import LottoList from "./src/components/LottoList";
+import LottoResult from "./src/components/LottoResult";
+import LottoShop from "./src/services/LottoShop";
+import { shallowEqual } from "./src/utils";
+
+const initialState = {
+  coin: null,
+  ticket: 0,
+  lottoList: [],
+  myLotto: [],
+};
+
+export default function App(app) {
+  this.app = app;
+  this.state = initialState;
+
+  new Header({ app });
+
+  new CoinInput({
+    app,
+    onClick: (coin, input) => {
+      try {
+        const lottoShop = LottoShop.getInstance()
+
+        lottoShop.coin = coin
+        
+        const lottoShopCoin = lottoShop.coin
+        const lottoShopTicket = lottoShop.ticket
+        const lottoShopLottoList = lottoShop.lottoList
```

Comment:
```suggestion
        const { coin, ticket, lottos } = lottoShop.buy(coin)
```

이런식으로 표현할 수 있도록 해주면 어떨까요?

Reply:
후... 왜 저 생각을 못 했을까요? ㅋㅋㅋㅋ